### PR TITLE
feat: add cra-todo-list-auth in examples

### LINKED ISSUE #209: Example: Todo app using CRA
# Feature request

## Is your feature request related to a problem? Please describe.

New example use case

## Describe the solution you'd like

- Create a initial React app in `examples/todos-react` using [CRA](https://create-react-app.dev/docs/getting-started)
- Build a todo list app, using Supabase as the backend. Use the [Slack Clone](https://github.com/supabase/supabase/blob/master/examples/slack-clone/README.md) as an example.
- Add a one-click deploy using Vercel deploy button

## Additional context

- We will create a Quick Start template in our Dashboard for this app
- Ant has done an example [here](https://github.com/supabase/supabase/tree/master/examples/todo), which is a good baseline
- I suggest this is done with pure JS, rather than TS
- We don't need auth in this one, just keep it simple
- This will be used by [swimm.io](https://swimm.io) to create a tutorial for Hacktoberfest
- Keep it as simple as possible - no additional libraries, minimal styling etc.

### PR DESCRIPTION
**Pull request by YashKandalkar**
## What kind of change does this PR introduce?

Adds a create-react-app todo example

## Additional context

closes #209 
So I mostly took help from the nextjs-todo-list example but I have added a password reset component too.
I have copy-pasta'd the readme from nextjs-todo and made minor changes in it for cra but haven't added a demo and deploy with vercel since it requires the folder to be in this repo (?)

Tell me if there are any changes to be made!

### CODE REVIEW

**Comment by YashKandalkar** [comment]
@kiwicopple I didn't knew if vercel deploys the sub folders so I have deployed it from my account: https://supabase.yashkandalkar.vercel.app/

File: examples/cra-todo-list-auth/.gitignore
Comment (ykdojo) [inline examples/cra-todo-list-auth/.gitignore:14]: Maybe add the .idea folder here to make it slightly simpler, too?

Comment (YashKandalkar) [reply examples/cra-todo-list-auth/.gitignore:14]: Will remove it 👍

**Review by ykdojo — APPROVED** [review]
Thanks! I think this is good to go unless @kiwicopple wants to check it before I merge it. If not, I'll probably merge it in 48 hours or so haha

**Comment by YashKandalkar** [comment]
@ykdojo Thank you for reviewing!  I've been watching your videos for years, they're awesome!

**Comment by ykdojo** [comment]
@YashKandalkar Thank you! Really appreciate you saying that 🙂

**Pull request merged**
